internal/api/v1beta1/app_types.go

```diff
@@ -501,6 +516,30 @@ func (s AppStatus) Condition(t AppConditionType) *AppCondition {
 	return nil
 }
 
+// DoCanary checks if canary deployment is needed for an app and gradually increases the traffic weight
+// based on the canary parameters provided by the users. Use it in app controller.
+func (app *App) DoCanary(now metav1.Time) {
```

Comment:
what if the given app is a large one? it can have 100 pods. How about scaling the pod number too?

Reply:
I am not sure if we should increase the number of Pod for the canary deployment too. Canary is a testing technique, I would not spin 100 pods for something I am not sure if it will work or will fail with time. If it succeeds then as a user I can increase the units.